carbon-design-system/carbon-addons-iot-react
# fix(TableHead): add id prop on TableExpandHeader

Closes #3485 

**Summary**

- fix the a11y violation by adding id prop on `TableExpandHeader`

According to the rule [H43: Using id and headers attributes to associate data cells with header cells in data tables](https://www.w3.org/TR/WCAG20-TECHS/H43) 

Header should use `id` prop for the any table cell to reference with. The default header for the first cell in `TableExpandRow` is set to `headers='expand'`. But we did not set an `id` on `TableExpandHeader` for the cell to refer to. 

The solution is to set ` id={`${tableId}-expand`} `on `TableExpandHeader` and `expandHeader={`${tableId}-expand`}` on `TableExpandRow`:


**Change List (commits, features, bugs, etc)**

-fix(TableBodyRow): add id prop on TableExpandHeader

**Acceptance Test (how to verify the PR)**

- go to table story [with row expansion](https://deploy-preview-3611--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--with-row-expansion)
- scan with IBM Equal Access Accessibility
- check no errors shows like `'The 'headers' attribute value "expand" does not reference a valid 'id' in this document'`

before:
![image](https://user-images.githubusercontent.com/24279794/196791121-6dc5e57b-5964-4ba3-b823-7c960fd9a6b4.png)

after:
![image](https://user-images.githubusercontent.com/24279794/196794365-1cb3780b-ebba-4d57-b7b4-7573ed61f29e.png)


**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here



**Things to look for during review**
